### DISCUSSION TITLE
Can I use survival analysis on an arbitrary regression problem (not survival) with censored outcome

### DISCUSSION THREAD

**Discussion by ogencoglu** [sebp/scikit-survival#475] [Q&A] [answered]
I have a machine learning problem of regression (not survival) where I have certain numerical and categorical input features and am trying to predict a numerical outcome. I know for a fact that the phenomena is highly non-linear. The problem has nothing to do with survival in that sense.

But for some of the outcomes (target values), I just know that they are below certain value. I don't know the exact value. Therefore I can not use standard scikit-learn regression models here as I can not calculate the loss accurately. Is survival analysis suitable for this use case?

Is there any fundamental mathematical or algorithmic difference between left-censoring and right-censoring in this context? For example, can I multiply my targets by -1 which will turn <20 to >-20 and switch from left-censoring to right censoring and the results will be the same?

Does it support negative target values just like regular regression? Or does survival analysis assume some physical meaning of "survival" and therefore negative target values does not make sense or anything like that?

In summary: how generalizable survival analysis is to arbitrary regression problems where there is some censoring in the targets?

**Comment by sebp** [comment, marked as answer]
Currently, scikit-survival only supports right censored events, that 
means when a lower bound on the true time of an event is known.

If you only have a upper bound on the true event time, this called left 
censoring. This is currently not supported, I'm afraid.

**Reply by ogencoglu** [reply]
Thanks for the reply. My problem does not even have the "time" or temporal concept as I explained. It is bunch of input features and a target value but the target value is coming from a sensor and it can not read below or above certain value. So, after all, survival analysis is not that generalizable to censored data.

**Comment by sebp** [comment]
Censoring is not restricted to analyzing time, it applies to any 
continues variable where you only have partial information.

Your example of having a senor that cannot measure below a certain 
threshold would indeed be an example for left censored data.

Another non time-related example would be the selling of online ads. 
Typically, the opportunity to display an ad goes to the highest bidder. 
Hence, if you are not the winner the price of the ad is right censored.
